src/ajax/xhr.js

```diff
@@ -19,7 +19,19 @@ jQuery.ajaxSettings.xhr = window.ActiveXObject !== undefined ?
 			// and http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9
 			// Although this check for six methods instead of eight
 			// since IE also does not support "trace" and "connect"
-			/^(get|post|head|put|delete|options)$/i.test( this.type ) &&
+			//
+			// Support: IE 10-11
+			// IE seems to error on cross-domain PATCH requests when ActiveX XHR
+			// is used. Use the following strategy:
+			// 1. If request is local, use the ActiveX XHR.
+			// 2. Otherwise, in IE9+ always use the native XHR.
+			// 3. Otherwise, in IE8 if request type is *not* included in the following
+			//    list, use the ActiveX XHR.
+			// 4. Otherwise, use the native XHR.
+			// jshint -W018
+			( !( document.documentMode < 9 ) ||
+			// jshint +W018
+				/^(get|post|head|put|delete|options)$/i.test( this.type ) ) &&
```

Comment:
From the top of my head, I don't think there is a gain... but it may have been lost in the mysts of ajax refactoring.

The sad truth is you can disable XMLHttpRequest and/or ActiveX and there are lots of subtle differences. I'm not certain they are all documented and I'm not convinced there is a proper all-encompassing priority to be found here.

Seeing as it's getting to this kind of browser-sniffing / test-with-a-gigantic-regexp situation, I wonder if we wouldn't be better served by only providing a default implementation that creates a native XHR and then document that you can change the factory to use ActiveX on a per-request basis. That's the whole point of having the factory as a setting.